src/AutoMerge/new-version.jl

```diff
@@ -46,57 +47,49 @@ function pull_request_build(data::GitHubAutoMergeData, ::NewVersion)::Nothing
         this_pr_can_use_special_jll_exceptions = false
     end
 
-    guidelines = Guideline[]
-    if !this_is_jll_package && pr_author_login ∉ data.authorized_authors
-        push!(guidelines,
-              guideline_jll_only_authorization) # 0
-    end
-
-    push!(guidelines,
-          guideline_pr_only_changes_allowed_files) # 1
+    jll_only_authorization = (!this_is_jll_package
+                              && pr_author_login ∉ data.authorized_authors)
```

Comment:
I don't really understand the meaning of this variable; it seems negated from what I'd expect `jll_only_authorization` to mean. Or maybe I'm just not thinking clearly 😅. Mind explaining it a little?

Reply:
It means that the PR is for a non-jll package but the author only has authorization for jll packages. I'm open for ways to make it clearer, if someone can think of something better.

Reply:
Ah ok. I think this one is more complicated because the check is being done in `applicable` instead of in the guideline. Maybe that could be addressed here?

Reply:
Another option would be to exclude this authorization failure from the guideline checking and immediately return when it has been detected. That's done for other authorization failures.

Reply:
That sounds fine. It would make the behavior consistent for all authorization failures.

Reply:
> It means that the PR is for a non-jll package but the author only has authorization for jll packages. I'm open for ways to make it clearer, if someone can think of something better.

How about just add a comment above this line with exactly what you wrote here? I think that's sufficient.